src/observable/IteratorObservable.ts

```diff
@@ -1,23 +1,32 @@
 import { root } from '../util/root';
 import { IScheduler } from '../Scheduler';
-import { Observable } from '../Observable';
+import { Observable, ArrayOrIterable } from '../Observable';
 import { iterator as Symbol_iterator } from '../symbol/iterator';
 import { TeardownLogic } from '../Subscription';
 import { Subscriber } from '../Subscriber';
+import { isArrayLike } from '../util/isArrayLike';
+
+export interface IteratorObservableState<T> {
```

Comment:
I think we may need, I recall some of dispatch is private and interfaces are hidden already. No need to expose unnecessary interface.